demo/src/core/intersection-observer-component.js

```diff
@@ -1,46 +1,72 @@
-import { onIntersecting, parseHtml } from './html-utils';
+import { html, LitElement } from 'lit';
+
+
+const DEFAULT_INT_OPTIONS = { root: null, rootMargin: '0px', threshold: 0.1 };
+
+/**
+ * Attach an Intersection Observer to a target element and execute a callback when it becomes visible.
+ *
+ * @param {Element} target - The target element to observe.
+ * @param {Function} callback - The callback function to execute when the target is intersecting.
+ * @param {Object} options - (Optional) Options to configure the Intersection Observer.
+ * @param {Element} [options.root=null] - The element that is used as the viewport for checking visibility.
+ * @param {string} [options.rootMargin='0px'] - Margin around the root. Can have values similar to CSS margin property.
+ * @param {number} [options.threshold=0.1] - The threshold at which the callback will be triggered.
+ */
+const onIntersecting = (target, callback, options = DEFAULT_INT_OPTIONS) => {
+  new IntersectionObserver((entries) => {
+    entries.forEach(entry => {
+      if (entry.isIntersecting) {
+        callback();
+        // observer.unobserve(entry.target);
+      }
+    });
+  }, options).observe(target);
+};
 
-let INTERSECTION_OBSERVER_ID = 0;
 
 /**
- * A simple intersection observer component that can be attached to a DOM element. This
- * component triggers a callback whenever it comes into view.
+ * This web component acts as a sentinel for an Intersection Observer,
+ * dispatching an 'intersecting' event when the observed element intersects
+ * with the view.
+ *
+ * @element intersection-observer
+ * @csspart sentinel - the sentinel element.
  *
- * @class
+ * @fires IntersectionObserverComponent#intersecting - Dispatched when the observed element intersects with the view.
  */
-class IntersectionObserverComponent {
-  /**
-   * The intersection observer component element reference.
-   *
-   * @private
-   * @type {Element}
-   */
-  #el;
-
-  /**
-   * Creates a new intersection observer component and attaches it to the DOM through the
-   * provided attach method.
-   *
-   * @constructor
-   * @param {(node?: Node) => void} attach - Callback function to customize
-   *     how the intersection observer is attached to document. Receives the intersection observer
-   *     HTML node as an argument.
-   * @param {function} intersectCallback - This function is called when the intersection observer
-   *     element is intersecting the viewport.
-   */
-  constructor(attach, intersectCallback) {
-    const id = `intersection-observer-${INTERSECTION_OBSERVER_ID += 1}`;
-
-    attach(parseHtml(`<div id="${id}" class="intersection-observer"></div>`)[0]);
-
-    this.#el = document.getElementById(id);
-
-    onIntersecting(this.#el, intersectCallback);
+export class IntersectionObserverComponent extends LitElement {
+
+  constructor() {
+    super();
+  }
```

Comment:
Refer to https://github.com/SRGSSR/pillarbox-web/pull/141/files#r1429898660

Reply:
Removed